test/evp_test.c

```diff
@@ -2042,7 +2045,9 @@ static int kdf_test_ctrl(EVP_TEST *t, EVP_KDF_CTX *kctx,
     if (!TEST_ptr(name = OPENSSL_strdup(value)))
         return 0;
     p = strchr(name, ':');
-    if (p != NULL)
+    if (p == NULL)
+        p = "";
+    else
         *p++ = '\0';
```

Comment:
Not sure about this. There may be value in distinguishing between "value not given" (NULL) and the empty value. What I would do is to modify `strlen(p)` to `p == NULL ? 0 : strlen(p)` below.

Reply:
Done.  It doesn't look like it cascades forwards, although I did reverse the order of the later conditions that checked `p != NULL` just in case.